File: ruby-1-uzd/Trikampis/main.rb
# frozen_string_literal: true

# Description/Explanation of Trikampis class
class Trikampis
  attr_accessor :trikmapio_tipas
  def initialize(put1, put2, put3)
    @put1 = put1
    @put2 = put2
    @put3 = put3
    @trikmapio_tipas = triangle
  end

  def pusperimetris
    (@put1 + @put2 + @put3) / 2
  end

  def trikampio_plotas(pusperimetris)
    Math.sqrt(
      pusperimetris * (pusperimetris - @put1) *
      (pusperimetris - @put2) * (pusperimetris - @put3)
    )
  end

  def triangle
    return 'Trikampis lygiakrastis' if lygiakrastis?
    return 'Trikampis lygiasonis' if lygiasonis?
    return 'Ivairekrastis' if ivairiakrastis?

    'Trinkampis neimanomas'
  end

  def lygiakrastis?
    @put1 == @put2 && @put2 == @put3 && @put3 == @put1
  end

  def lygiasonis?
    @put1 == @put2 || @put2 == @put3 || @put1 == @put3
  end

  def ivairiakrastis?
    @put1 != @put2 && @put2 != @put3 && @put1 != @put3
  end
end

puts 'Ivesk krastiniu ilgius'
put1 = gets.to_f
put2 = gets.to_f
put3 = gets.to_f

trikampis = Trikampis.new(put1, put2, put3)
trikampis.pusperimetris
puts "Trikampio pusperimetris yra : #{trikampis.pusperimetris}"
trikampis.trikampio_plotas(trikampis.pusperimetris)
puts "Plotas : #{trikampis.trikampio_plotas(trikampis.pusperimetris)}"

File: ruby-1-uzd/Trikampis/main.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main()
{
    int a,b,c;
    double plotas,p;
    cout <<"ivesk krastines"<<endl;
    cin >>a>>b>>c;
    p=(a+b+c)/2;
    //ar trikampis egzistuoja
    if (a<=0 || b<=0 || b<=0)  cout<<"ramiakas, neigiamu grastiniu but negali";
    else if (a+b>c && b+c>a && a+c>b){
        cout <<"trikampis egzistuoja"<<endl;
        //koks trikampis yra
        plotas = sqrt(p*(p-a)*(p-b)*(p-c));
        cout <<"Plotas = "<<plotas<<endl;
        if (a==b && b==c && c==a ) cout <<"trikampis lygiakrastis"<<endl;
        else if (a==b || b==c || a==c ) cout <<"trikampis lygiasonis"<<endl;
        else if (a!=b && b!=c && a!=c) cout <<"trikampis ivairiakrastis"<<endl;

        }
    else cout <<"trikampis neegzistuoja"<<endl;
    return 0;
}
